# London Housing Data Analysis

In [1]:
# Load Packages
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings("ignore")

# Load Data Frame

In [2]:
london_housing = pd.read_csv("5. London Housing Data.csv")
london_housing.head()

date            area  average_price       code  houses_sold  \
0  1/1/1995  city of london          91449  E09000001         17.0   
1  2/1/1995  city of london          82203  E09000001          7.0   
2  3/1/1995  city of london          79121  E09000001         14.0   
3  4/1/1995  city of london          77101  E09000001          7.0   
4  5/1/1995  city of london          84409  E09000001         10.0   

   no_of_crimes  
0           NaN  
1           NaN  
2           NaN  
3           NaN  
4           NaN

# Data PreProcessing

In [3]:
london_housing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13549 entries, 0 to 13548
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   date           13549 non-null  object 
 1   area           13549 non-null  object 
 2   average_price  13549 non-null  int64  
 3   code           13549 non-null  object 
 4   houses_sold    13455 non-null  float64
 5   no_of_crimes   7439 non-null   float64
dtypes: float64(2), int64(1), object(3)
memory usage: 635.2+ KB


In [4]:
london_housing.isnull().sum()

date                0
area                0
average_price       0
code                0
houses_sold        94
no_of_crimes     6110
dtype: int64

In [5]:
# Convert Date
london_housing["date"] = pd.to_datetime(london_housing["date"], dayfirst=True)

# 1-i. Add a new column 'year' in the data frame, which contains years only.

In [6]:
london_housing["year"] = london_housing["date"].dt.year
london_housing.head()

date            area  average_price       code  houses_sold  \
0 1995-01-01  city of london          91449  E09000001         17.0   
1 1995-01-02  city of london          82203  E09000001          7.0   
2 1995-01-03  city of london          79121  E09000001         14.0   
3 1995-01-04  city of london          77101  E09000001          7.0   
4 1995-01-05  city of london          84409  E09000001         10.0   

   no_of_crimes  year  
0           NaN  1995  
1           NaN  1995  
2           NaN  1995  
3           NaN  1995  
4           NaN  1995

# 1-ii. Add a new column 'year' in the data frame, which contains years only.

In [7]:
london_housing["month"] = london_housing["date"].dt.month
london_housing.head()

date            area  average_price       code  houses_sold  \
0 1995-01-01  city of london          91449  E09000001         17.0   
1 1995-01-02  city of london          82203  E09000001          7.0   
2 1995-01-03  city of london          79121  E09000001         14.0   
3 1995-01-04  city of london          77101  E09000001          7.0   
4 1995-01-05  city of london          84409  E09000001         10.0   

   no_of_crimes  year  month  
0           NaN  1995      1  
1           NaN  1995      1  
2           NaN  1995      1  
3           NaN  1995      1  
4           NaN  1995      1

# 2. Remove the columns 'year' and 'month' from the dataframe.

In [8]:
london_housing.drop(["year", "month"], axis=1, inplace=True)
london_housing.head()

date            area  average_price       code  houses_sold  \
0 1995-01-01  city of london          91449  E09000001         17.0   
1 1995-01-02  city of london          82203  E09000001          7.0   
2 1995-01-03  city of london          79121  E09000001         14.0   
3 1995-01-04  city of london          77101  E09000001          7.0   
4 1995-01-05  city of london          84409  E09000001         10.0   

   no_of_crimes  
0           NaN  
1           NaN  
2           NaN  
3           NaN  
4           NaN

# 3. Show all the records where 'No. of Crimes' is 0. And, how many such records are there?

In [9]:
(london_housing["no_of_crimes"] == 0).sum()

np.int64(104)

In [10]:
london_housing[london_housing["no_of_crimes"] == 0]

date            area  average_price       code  houses_sold  \
72  2001-01-01  city of london         284262  E09000001         24.0   
73  2001-01-02  city of london         198137  E09000001         37.0   
74  2001-01-03  city of london         189033  E09000001         44.0   
75  2001-01-04  city of london         205494  E09000001         38.0   
76  2001-01-05  city of london         223459  E09000001         30.0   
..         ...             ...            ...        ...          ...   
178 2009-01-11  city of london         397909  E09000001         11.0   
179 2009-01-12  city of london         411955  E09000001         16.0   
180 2010-01-01  city of london         464436  E09000001         20.0   
181 2010-01-02  city of london         490525  E09000001          9.0   
182 2010-01-03  city of london         498241  E09000001         15.0   

     no_of_crimes  
72            0.0  
73            0.0  
74            0.0  
75            0.0  
76            0.0  
..            ...  
178           0.0  
179           0.0  
180           0.0  
181           0.0  
182           0.0  

[104 rows x 6 columns]

# 4. What is the maximum & minimum 'average_price' per year in england?

In [11]:
# Adding new coloum
london_housing["year"] = london_housing["date"].dt.year

# Filtering 'England'
england = london_housing[london_housing["area"] == "england"]

# Minimum and Maximum value per year in England
min_max = england.groupby("year")["average_price"].agg(["min", "max"])

min_max

min     max
year                
1995   52788   53901
1996   52333   55755
1997   55789   61564
1998   61659   65743
1999   65522   75071
2000   75219   84191
2001   84245   95992
2002   96215  119982
2003  121610  138985
2004  139719  160330
2005  158572  167244
2006  166544  182031
2007  181824  194764
2008  165795  191750
2009  159340  174136
2010  174458  180807
2011  173046  177335
2012  174161  180129
2013  176816  188544
2014  188265  203639
2015  202856  219582
2016  220361  231922
2017  231593  242628
2018  240428  248620
2019  243281  250410
2020  247355  247355

# 5. What is the Maximum & Minimum No. of Crimes recorded per area?

In [12]:
min_max = london_housing.groupby("area")[ "no_of_crimes"].agg(["min", "max"])

min_max

min     max
area                                  
barking and dagenham    1217.0  2049.0
barnet                  1703.0  2893.0
bexley                   860.0  1914.0
brent                   1850.0  2937.0
bromley                 1441.0  2637.0
camden                  2079.0  4558.0
city of london             0.0    10.0
croydon                 2031.0  3263.0
ealing                  1871.0  3401.0
east midlands              NaN     NaN
east of england            NaN     NaN
enfield                 1635.0  2798.0
england                    NaN     NaN
greenwich               1513.0  2853.0
hackney                 1870.0  3466.0
hammersmith and fulham  1323.0  2645.0
haringey                1536.0  3199.0
harrow                   937.0  1763.0
havering                1130.0  1956.0
hillingdon              1445.0  2819.0
hounslow                1529.0  2817.0
inner london               NaN     NaN
islington               1871.0  3384.0
kensington and chelsea  1347.0  2778.0
kingston upon thames     692.0  1379.0
lambeth                 2381.0  4701.0
lewisham                1675.0  2813.0
london                     NaN     NaN
merton                   819.0  1623.0
newham                  2130.0  3668.0
north east                 NaN     NaN
north west                 NaN     NaN
outer london               NaN     NaN
redbridge               1487.0  2560.0
richmond upon thames     700.0  1551.0
south east                 NaN     NaN
south west                 NaN     NaN
southwark               2267.0  3821.0
sutton                   787.0  1425.0
tower hamlets           1646.0  3316.0
waltham forest          1575.0  2941.0
wandsworth              1582.0  3051.0
west midlands              NaN     NaN
westminster             3504.0  7461.0
yorks and the humber       NaN     NaN

# 6. Show the total count of records of each area, where average price is less than 100000.

In [13]:
# Filter rows where average_price < 100000
london_housing[london_housing["average_price"] < 100000]["area"].value_counts()

area
north east              112
north west              111
yorks and the humber    110
east midlands            96
west midlands            94
england                  87
barking and dagenham     85
south west               78
east of england          76
newham                   72
bexley                   64
waltham forest           64
lewisham                 62
havering                 60
south east               59
greenwich                59
croydon                  57
sutton                   54
enfield                  54
hackney                  53
redbridge                52
southwark                48
tower hamlets            47
outer london             46
hillingdon               44
hounslow                 41
lambeth                  41
brent                    40
london                   39
merton                   35
haringey                 33
bromley                  33
ealing                   31
inner london             31
harrow                   30
kingston upon t

# Conclusion

1. The dataset contains 13,549 records with details on housing prices, areas, houses sold, and crime rates. 
   Some columns, like `houses_sold` and `no_of_crimes`, have missing values.

2. A new `year` column was created for easier time-based analysis, while unnecessary columns were removed.

3. Records with zero crimes were identified and counted separately.

4. Yearly maximum and minimum average housing prices in England were analyzed, showing how property values 
   changed over time.

5. The maximum and minimum number of crimes were recorded for each area, providing insight into 
   regional safety differences.

6. Areas with records showing average prices below 100,000 were identified, highlighting affordability zones.

Overall, the analysis reveals housing price trends, regional crime statistics, and affordability insights. 
It provides a solid foundation for further exploration, such as predicting housing price trends or 
studying correlations between crime and property values.